packages/launch/src/hooks/use-domain-search.ts

```diff
@@ -3,37 +3,48 @@
  */
 import { useSelect } from '@wordpress/data';
 import { useDispatch } from '@wordpress/data';
-import { isGoodDefaultDomainQuery } from '@automattic/domain-picker';
+import { isGoodDefaultDomainQuery, DOMAIN_QUERY_MINIMUM_LENGTH } from '@automattic/domain-picker';
 
 /**
  * Internal dependencies
  */
 import { LAUNCH_STORE } from '../stores';
 import { useSite, useTitle } from './';
-import { isDefaultSiteTitle } from '../utils';
 import { useSiteDomains } from './use-site-domains';
 
 export function useDomainSearch(): {
 	domainSearch: string;
 	isLoading: boolean;
 	setDomainSearch: ( search: string ) => void;
 } {
-	const { domainSearch } = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
-	const { title } = useTitle();
-	const { isLoadingSite } = useSite();
+	const existingDomainSearch = useSelect(
+		( select ) => select( LAUNCH_STORE ).getDomainSearch(),
+		[]
+	);
+	const { isDefaultTitle, title } = useTitle();
+	const { isLoadingSite: isLoading } = useSite();
 	const { siteSubdomain } = useSiteDomains();
 
 	const { setDomainSearch } = useDispatch( LAUNCH_STORE );
 
-	let search = domainSearch.trim() || filterUnsuitableTitles( title || '' );
+	const searchQueryFromTitle =
+		typeof title !== 'undefined' && title.length >= DOMAIN_QUERY_MINIMUM_LENGTH && ! isDefaultTitle
```

Comment:
With the latest changes, If title is `undefined`:

- `isDefaultTitle` will always be `true`, and therefore `! isDefaultTitle` will always be `false`
- the point above clashes with the `typeof title !== 'undefined'` check